### DISCUSSION TITLE
Just some DB schema based on web services and scraped data.

### DISCUSSION THREAD

**Discussion by Grrrolf** [idolpx/csdb-ng#9] [Show and tell]
This db schema was created to store data, not to run a CSDb system. Mainly used this for analytics and data cleansing on CSDb.

I think this information is an interesting way to get to understand what data is freely available right now and lots is missing.

![csdb_webservices_and_such](https://user-images.githubusercontent.com/10052794/170771600-260f51f9-db8d-4815-836b-a7b57f80cd31.png)

**Comment by Grrrolf** [comment]
The description look-up tables data (data may be old):

**release_type_description**
| id | description | creation\_date | short\_description | last\_updated\_on |
| :--- | :--- | :--- | :--- | :--- |
| 1 | C64 Demo | 2020-01-28 20:44:17.000000 | Demo | 2020-01-28 20:45:16.829272 |
| 2 | C64 One-File Demo | 2020-01-28 20:44:17.000000 | Single file demo | 2020-01-28 20:45:16.864573 |
| 3 | C64 Intro | 2020-01-28 20:44:17.000000 | Intro demo | 2020-01-28 20:45:16.883892 |
| 4 | C64 4K Intro | 2020-01-28 20:44:17.000000 | 4k demo | 2020-01-28 20:45:16.902279 |
| 5 | C64 Crack Intro | 2020-01-28 20:44:17.000000 | Crack Intro | 2020-01-28 20:45:16.922632 |
| 6 | REU Release | 2020-01-28 20:44:17.000000 | REU | 2020-01-28 20:45:16.942304 |
| 7 | C64 Music | 2020-01-28 20:44:17.000000 | Music | 2020-01-28 20:45:16.959765 |
| 8 | C64 Music Collection | 2020-01-28 20:44:17.000000 | Music collection | 2020-01-28 20:45:16.977567 |
| 9 | C64 Graphics | 2020-01-28 20:44:17.000000 | GFX | 2020-01-28 20:45:16.998460 |
| 10 | C64 Graphics Collection | 2020-01-28 20:44:17.000000 | GFX collection | 2020-01-28 20:45:17.019083 |
| 11 | C64 Game | 2020-01-28 20:44:17.000000 | Game | 2020-01-28 20:45:17.036352 |
| 12 | C64 32K Game | 2020-01-28 20:44:17.000000 | 32k game | 2020-01-28 20:45:17.059004 |
| 13 | C64 Diskmag | 2020-01-28 20:44:17.000000 | Diskmag | 2020-01-28 20:45:17.080000 |
| 14 | C64 Charts | 2020-01-28 20:44:17.000000 | Charts | 2020-01-28 20:45:17.100072 |
| 15 | C64 Tool | 2020-01-28 20:44:17.000000 | Tool | 2020-01-28 20:45:17.117605 |
| 16 | C64 Invitation | 2020-01-28 20:44:17.000000 | Invitation | 2020-01-28 20:45:17.137162 |
| 17 | C64 Misc. | 2020-01-28 20:44:17.000000 | Misc | 2020-01-28 20:45:17.157988 |
| 18 | C64 1K Intro | 2020-01-28 20:44:17.000000 | 1k Intro | 2020-01-28 20:45:17.174051 |
| 19 | C64 Game Preview | 2020-01-28 20:44:17.000000 | Game preview | 2020-01-28 20:45:17.191275 |
| 20 | C64 Crack | 2020-01-28 20:44:17.000000 | Crack | 2020-01-28 20:45:17.206281 |
| 21 | Other Platform C64 Tool | 2020-01-28 20:44:17.000000 | Other platform C64 tool | 2020-01-28 20:45:17.223722 |
| 22 | C64 Basic Demo | 2020-01-28 20:44:17.000000 | Basic demo | 2020-01-28 20:45:17.242150 |
| 23 | C64 Hardware | 2020-01-28 20:44:17.000000 | Hardware | 2020-01-28 20:45:17.258463 |
| 24 | C64 Fake Demo | 2020-01-28 20:44:17.000000 | Fake demo | 2020-01-28 20:45:17.280112 |
| 25 | C64 Tool Collection | 2020-01-28 20:44:17.000000 | Tool collection | 2020-01-28 20:45:17.301690 |
| 26 | C64 Papermag | 2020-01-28 20:44:17.000000 | Papermag | 2020-01-28 20:45:17.317221 |
| 27 | C128 Release | 2020-01-28 20:44:17.000000 | C128 | 2020-01-28 20:45:17.333479 |
| 28 | SuperCPU Release | 2020-01-28 20:44:17.000000 | SuperCPU | 2020-01-28 20:45:17.357059 |
| 29 | C64 1K Game | 2020-01-28 20:44:17.000000 | 1k game | 2020-01-28 20:45:17.375185 |
| 30 | C64 2K Game | 2020-01-28 20:44:17.000000 | 2k game | 2020-01-28 20:45:17.393546 |
| 31 | C64 512b Game | 2020-01-28 20:44:17.000000 | 512b game | 2020-01-28 20:45:17.412041 |
| 33 | C64 Disk Cover | 2020-01-28 20:44:17.000000 | Disk cover | 2020-01-28 20:45:17.446089 |
| 34 | C64 Votesheet | 2020-01-28 20:44:17.000000 | Votesheet | 2020-01-28 20:45:17.466551 |
| 35 | C64 4K Game | 2020-01-28 20:44:17.000000 | 4k game | 2020-01-28 20:45:17.484412 |
| 36 | C64 256b Intro | 2020-01-28 20:44:17.000000 | 256b intro | 2020-01-28 20:45:17.504078 |
| 37 | IDE64 Release | 2020-01-28 20:44:17.000000 | IDE64 | 2020-01-28 20:45:17.558499 |
| 40 | C64 DTV | 2020-01-28 20:44:17.000000 | DTV | 2020-01-28 20:45:17.575722 |
| 41 | C64 Fake Game | 2020-01-28 20:44:17.000000 | Fake game | 2020-01-28 20:45:17.591971 |
| 42 | BBS software | 2020-01-28 20:44:17.000000 | BBS | 2020-01-28 20:45:17.646233 |
| 43 | BBS Graphics | 2020-01-28 20:44:17.000000 | BBS gfx | 2020-01-28 20:45:17.692647 |
| 44 | C64 Intro Collection | 2020-01-28 20:44:17.000000 | Intro collection | 2020-01-28 20:45:17.610878 |
| 45 | Related Release | 2020-01-28 20:44:17.000000 | Related | 2020-01-28 20:45:17.628375 |
| 46 | EasyFlash Release | 2020-01-28 20:44:17.000000 | Easy Flash | 2020-01-28 20:45:17.671586 |

**event_compo_type_description**
| compo\_type\_id | description |
| :--- | :--- |
| 0 | Outside Compo |
| 1 | C64 Demo |
| 2 | C64 4K Intro |
| 3 | C64 Music |
| 4 | C64 Graphics |
| 5 | C64 Game |
| 6 | C64 32K Game |
| 7 | WiLd Demo |
| 8 | C64 1K Intro |
| 9 | C64 Basic Demo |
| 10 | Mixed |
| 11 | C64 512B Game |
| 12 | C64 1K Game |
| 13 | C64 2K Game |
| 14 | C64 Fast Intro |
| 15 | C64 Disk Cover |
| 16 | C64 4K Game |
| 17 | C64 Sample Music |
| 18 | C64 256b Intro |
| 19 | C64 64k Demo |
| 20 | C64 2SID |
| 21 | Mixed Graphics |
| 22 | Mixed Music |
| 23 | Mixed Demo |
| 24 | Mixed Game |

**event_type_description**
| id | event\_type | creation\_date |
| :--- | :--- | :--- |
| 1 | Copy Parties | 2019-04-07 19:43:12 |
| 2 | Demo Parties | 2019-04-07 19:43:12 |
| 3 | Meetings | 2019-04-07 19:43:12 |
| 4 | Internal Meetings | 2019-04-07 19:43:12 |
| 5 | C64 Only Parties | 2019-04-07 19:43:12 |
| 6 | 8 bit Parties | 2019-04-07 19:43:12 |
| 8 | Standalone Compos | 2019-04-07 19:43:12 |

**release_credit_type_description**
| id | credit\_description |
| :--- | :--- |
| 1 | Code |
| 2 | Music |
| 3 | Graphics |
| 4 | Design |
| 5 | Idea |
| 6 | Storyboard |
| 7 | Disk Cover |
| 8 | Crack |
| 9 | Trainer |
| 10 | Text |
| 11 | Import |
| 12 | Original Supply |
| 13 | Charset |
| 14 | NTSC-Fix |
| 15 | PAL-Fix |
| 16 | Linking |
| 17 | Concept |
| 18 | Bug-Fix |
| 19 | Loader |
| 20 | Vote Sheet |
| 21 | Paper Art |
| 22 | Docs |
| 23 | Ripping |
| 24 | Interviewed |
| 25 | Test |
| 26 | Help |
| 27 | Sampling |
| 28 | BBS Graphics |
| 29 | Translation |
| 30 | ----------- |
| 31 | Directory Art |

**scener_credit_type_description**
| credit\_type\_id | description | data\_added\_on | data\_last\_updated\_on |
| :--- | :--- | :--- | :--- |
| 1 | Code | 2019-07-23 20:55:16.998694 | NULL |
| 2 | Music | 2019-07-23 20:55:17.015332 | NULL |
| 3 | Graphics | 2019-07-23 20:55:17.025668 | NULL |
| 4 | Design | 2019-07-23 20:55:17.039530 | NULL |
| 5 | Idea | 2019-07-23 20:55:17.053612 | NULL |
| 6 | Storyboard | 2019-07-23 20:55:17.063715 | NULL |
| 7 | Disk Cover | 2019-07-23 20:55:17.089591 | NULL |
| 8 | Crack | 2019-07-23 20:55:17.110060 | NULL |
| 9 | Trainer | 2019-07-23 20:55:17.130554 | NULL |
| 10 | Text | 2019-07-23 20:55:17.147583 | NULL |
| 11 | Import | 2019-07-23 20:55:17.162505 | NULL |
| 12 | Original Supply | 2019-07-23 20:55:17.183834 | NULL |
| 13 | Charset | 2019-07-23 20:55:17.196884 | NULL |
| 14 | NTSC-Fix | 2019-07-23 20:55:17.211544 | NULL |
| 15 | PAL-Fix | 2019-07-23 20:55:17.229161 | NULL |
| 16 | Linking | 2019-07-23 20:55:17.246931 | NULL |
| 17 | Concept | 2019-07-23 20:55:17.259265 | NULL |
| 18 | Bug-Fix | 2019-07-23 20:55:17.272774 | NULL |
| 19 | Loader | 2019-07-23 20:55:17.286098 | NULL |
| 20 | Vote Sheet | 2019-07-23 20:55:17.302095 | NULL |
| 21 | Paper Art | 2019-07-23 20:55:17.320120 | NULL |
| 22 | Docs | 2019-07-23 20:55:17.335515 | NULL |
| 23 | Ripping | 2019-07-23 20:55:17.347826 | NULL |
| 24 | Interviewed | 2019-07-23 20:55:17.359861 | NULL |
| 25 | Test | 2019-07-23 20:55:17.371760 | NULL |
| 26 | Help | 2019-07-23 20:55:17.380318 | NULL |
| 27 | Sampling | 2019-07-23 20:55:17.393556 | NULL |
| 28 | BBS Graphics | 2019-07-23 20:55:17.406747 | NULL |
| 29 | Translation | 2019-07-23 20:55:17.419469 | NULL |
| 31 | Directory Art | 2019-07-23 20:55:17.428461 | NULL |

**scener_profession_type_description**
| profession\_id | description | creation\_date |
| :--- | :--- | :--- |
| 1 | Organizer | 2019-04-07 19:43:12.000000 |
| 2 | Co-Organizer | 2019-04-07 19:43:12.000000 |
| 3 | Coder | 2019-04-07 19:43:12.000000 |
| 4 | Graphician | 2019-04-07 19:43:12.000000 |
| 5 | Fullscreen Graphician | 2019-04-07 19:43:12.000000 |
| 6 | Logo Graphician | 2019-04-07 19:43:12.000000 |
| 7 | Musician | 2019-04-07 19:43:12.000000 |
| 8 | Swapper | 2019-04-07 19:43:12.000000 |
| 9 | Mega Swapper | 2019-04-07 19:43:12.000000 |
| 10 | Modem Trader | 2019-04-07 19:43:12.000000 |
| 11 | Net Trader | 2019-04-07 19:43:12.000000 |
| 12 | FTP Administrator | 2019-04-07 19:43:12.000000 |
| 13 | FTP Supporter | 2019-04-07 19:43:12.000000 |
| 14 | Webmaster | 2019-04-07 19:43:12.000000 |
| 15 | NTSC-Fixer | 2019-04-07 19:43:12.000000 |
| 16 | Cracker | 2019-04-07 19:43:12.000000 |
| 17 | Original Supplier | 2019-04-07 19:43:12.000000 |
| 18 | Diskmag Editor | 2019-04-07 19:43:12.000000 |
| 19 | Webzine Editor | 2019-04-07 19:43:12.000000 |
| 20 | Cover Designer | 2019-04-07 19:43:12.000000 |
| 21 | Author | 2019-04-07 19:43:12.000000 |
| 22 | Sysop | 2019-04-07 19:43:12.000000 |
| 23 | Co-Sysop | 2019-04-07 19:43:12.000000 |
| 24 | Hardware-Guru | 2019-04-07 19:43:12.000000 |
| 25 | PAL-Fixer | 2019-04-07 19:43:12.000000 |
| 26 | Hacker | 2019-04-07 19:43:12.000000 |
| 27 | Phreaker | 2019-04-07 19:43:12.000000 |
| 28 | Public Relations Manager | 2019-04-07 19:43:12.000000 |
| 29 | Importer | 2019-04-07 19:43:12.000000 |
| 30 | Papermag Editor | 2019-04-07 19:43:12.000000 |
| 31 | BBS Graphician | 2019-04-07 19:43:12.000000 |
| 32 | Translator | 2019-04-07 19:43:12.000000 |
| 33 | Tester | 2019-04-07 19:43:12.000000 |

**group_types_description**
| id | description | creation\_date |
| :--- | :--- | :--- |
| 1 | Demo Group | 2019-04-07 19:43:12 |
| 2 | Cracker Group | 2019-04-07 19:43:12 |
| 3 | Music Group | 2019-04-07 19:43:12 |
| 4 | Graphics Group | 2019-04-07 19:43:12 |
| 5 | Hacker Group | 2019-04-07 19:43:12 |
| 6 | Magazine Staff | 2019-04-07 19:43:12 |
| 7 | Game Development Group | 2019-04-07 19:43:12 |
| 8 | Import Group | 2019-04-07 19:43:12 |
| 9 | Fixing Group | 2019-04-07 19:43:12 |
| 10 | Swapping Group | 2019-04-07 19:43:12 |
| 11 | Cover Group | 2019-04-07 19:43:12 |


**bbs_types_description**
| id | bbs\_type | creation\_date |
| :--- | :--- | :--- |
| 1 | Legal | 2019-04-07 19:43:12 |
| 2 | Illegal | 2019-04-07 19:43:12 |
| 3 | hp | 2019-04-07 19:43:12 |
| 4 | Wares | 2019-04-07 19:43:12 |

**Comment by idolpx** [comment]
Nice!  Good info!  Thanks for sharing.
